addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/MiHomeGatewayHandler.java

```diff
@@ -0,0 +1,575 @@
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing MiHomeGatewayHandler...");
+        Configuration config = getConfig();
+        if (verifyThingConfiguration(config)) {
+            createApiManager();
+            if (!isGatewayThingAlreadyRegistered(thing)) {
+                logger.debug("About to register a new gateway");
+                registerNewGateway();
+            } else {
+                logger.warn(ALREADY_REGISTERED_GATEWAY);
+                updateStatus(ThingStatus.REMOVED);
+            }
+            scheduleUpdateTask(this.updateInterval);
+        }
+    }
+
+    /** Checks if there is already a registered thing for a given gateway device */
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    private boolean isGatewayThingAlreadyRegistered(Thing thing) {
+        ServiceReference ref = bundleContext.getServiceReference(ThingRegistry.class);
+        thingRegistry = (ThingRegistry) bundleContext.getService(ref);
+        Collection<Thing> things = thingRegistry.getAll();
+        for (Thing registeredThing : things) {
+            ThingUID thingUID = registeredThing.getUID();
+            if (thingUID != null
+                    && registeredThing.getThingTypeUID().equals(MiHomeBindingConstants.THING_TYPE_GATEWAY)) {
+                Configuration config = registeredThing.getConfiguration();
+                String code = (String) config.get(MiHomeBindingConstants.CONFIG_GATEWAY_CODE);
+                if (gatewayCode.equals(code) && (!thing.getUID().equals(registeredThing.getUID()))) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public void registerNewGateway() {
+        JsonObject registrationResponse = apiManager.registerGateway(getThing().getLabel(), gatewayCode);
+        if (registrationResponse != null) {
+            JsonObject responseData = registrationResponse.get(JSONResponseConstants.DATA_KEY).getAsJsonObject();
+            gatewayId = responseData.get(DeviceConstants.DEVICE_ID_KEY).getAsInt();
+            setProperties(responseData);
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            logger.warn(NOT_SUCCESSFULL_REGISTRATION);
+        }
+    }
+
+    private void scheduleUpdateTask(long updateInterval) {
+        if (updateTaskResult != null) {
+            this.updateTaskResult.cancel(true);
+            this.updateTaskResult = null;
+        }
+        /** The refresh thread which is used for checking the "last_seen_at" property of the gateway */
+        Runnable refreshRunnable = new Runnable() {
+            @Override
+            public void run() {
+                logger.debug("Starting the refresh thread...");
+                checkGateway();
+            }
+        };
+        try {
+            scheduler.scheduleAtFixedRate(refreshRunnable, INITIAL_REFRESH_DELAY, updateInterval, TimeUnit.SECONDS);
+        } catch (Exception e) {
```

Comment:
It looks like the exceptions thrown by this method extend RuntimeException, which means they do not *have* to be caught. They also only relate to scheduling, so the error message logged is actually misleading, there wasn't a problem checking the gateway, there was a problem scheduling the check for the gateway.

Looking at the code, the command can never be null, and we should have other validation that the refresh period is positive (do we?). So the only exception that should be possible is an inability to schedule, which sounds like a system problem that should really just propagate.

I'd remove the exception handler from here, as its not adding any additional value.